doc/isis/frr-isis-sonic-yang-model-hld.md

```diff
@@ -0,0 +1,713 @@
+passive                            = bolean                ; Advertise the interface in the ISIS topology, but don't allow it to form adjacencies. Default "false"
+hello_padding                      = boolean               ; Add padding to ISIS hello PDUs
+network_type                       = "POINT_TO_POINT"/"BROADCAST"      ; ISIS interface type
+enable_bfd                         = boolean               ; Monitor ISIS peers on this interface
+bfd_profile                        = string                ; Let BFD use a pre-configured profile
+metric                             = UINT32                ; Metric value. Range 0..16777215. Default 0
+csnp_interval                      = boolean               ; Complete Sequence Number PDU (CSNP) generation interval. Range 1..600. Default 10 in seconds
+psnp_interval                      = boolean               ; Partial Sequence Number PDU (PSNP) generation interval. Range 1..120. Default 2 in seconds
+hello_interval                     = UINT32                ; Hello interval between consecutive hello messages. Range 1..600. Default 3 in seconds
+hello_multiplier                   = UINT16                ; Multiplier for the hello holding time. Range 2..100. Default 10
+authentication_key                 = string                ; Authentication password
+authentication_type                = "TEXT"/"MD5"         ; Authentication keychain type
+
+
+Tree view
+     +--rw ISIS_INTERFACE
+     |  +--rw ISIS_INTERFACE_LIST* [instance ifname]
+     |     +--rw instance                         string
+     |     +--rw ifname                           string
+     |     +--rw ipv4_routing_instance?           string
+     |     +--rw ipv6_routing_instance?           string
+     |     +--rw passive?                         boolean
+     |     +--rw hello_padding?                   boolean
+     |     +--rw network_type?                    circuit-type
+     |     +--rw enable_bfd?                      boolean
+     |     +--rw bfd_profile?                     string
+     |     +--rw metric?                          uint32
+     |     +--rw csnp_interval?                   uint16
+     |     +--rw psnp_interval?                   uint16
+     |     +--rw hello_interval?                  uint32
+     |     +--rw hello_multiplier?                uint16
+     |     +--rw authentication_key?              string
+     |     +--rw authentication_type?             AUTH_MODE
+```
+
+#### YANG Model Enhancements
+
+##### SONiC ISIS Global
+Global ISIS Yang container is sonic-isis.yang.
+
+```
+        container ISIS_GLOBAL {
+
+            list ISIS_GLOBAL_LIST {
+
+                max-elements "1";
+
+                key "vrf_name instance";
+
+                leaf vrf_name {
+                    type union {
+                        type string {
+                            pattern "default";
+                        }
+                        type leafref {
```

Comment:
Do we have enable ISIS support for user VRF as well? if yes, why do have max-elements 1 at line 225?

Reply:
This was added based on the FRR documentation mentioning  "isisd does not yet support multiple ISIS processes but you must specify the name of ISIS process" in https://docs.frrouting.org/en/latest/isisd.html